Src/GoogleApis.Auth.DotNet4/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -0,0 +1,255 @@
+                                        credentialPath,
+                                        e.Message));
+            }
+
+            // 3. Then try the compute engine, if not already checked         
+            if (!checkedComputeEngine)
+            {
+                Logger.Info("Checking whether the application is running on ComputeEngine.");
+                bool isRunningOnComputeEngine = ComputeCredential.IsRunningOnComputeEngine().Result;
+                checkedComputeEngine = true;
+                if (isRunningOnComputeEngine)
+                {
+                    Logger.Info("ComputeEngine check passed. Loading ComputeEngine Credentials.");
+                    return new ComputeCredential();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Converts a stream to a credential object.
+        /// </summary>
+        internal static ICredential LoadFromStream(Stream stream)
+        {
+            ClientCredentialParameters clientCredentialParameters = ClientCredentialParameters.Load(stream);
+          
+            switch (clientCredentialParameters.CredentialType)
+            {
+                case ClientCredentialType.UserCredential:
+                    try
+                    {
+                        var initializer = new GoogleAuthorizationCodeFlow.Initializer
+                        {
+                            ClientSecrets = clientCredentialParameters.ClientSecrets
+                        };
+
+                        TokenResponse token = new TokenResponse()
+                        {
+                            RefreshToken = clientCredentialParameters.RefreshToken
+                        };
+
+                        var flow = new GoogleAuthorizationCodeFlow(initializer);
+                        return new UserCredential(flow, "ApplicationDefaultCredentials", token);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                case ClientCredentialType.ServiceAccountCredential:
+                    try
+                    {
+                        var initializer = new ServiceAccountCredential.Initializer(clientCredentialParameters);
+                        return new ServiceAccountCredential(initializer);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                default:
+                    throw new InvalidOperationException("Error reading credentials from stream, 'type' field not specified.");
+            }
+        }
+
+        /// <summary> 
+        /// Helper method to return well known credential file path. This file is created by 'gcloud auth login'.
```

Comment:
I would be wary of adding a direct link to the documentation, as the doc team might restructure the url. 

The very first link when you search for 'gcloud auth login' points to the documentation.